main_2024.py

```diff
@@ -374,7 +412,16 @@ def main() -> int:
                 main_logger.debug(
                     "geolocation confidence: " + str(detection_world.confidence), True
                 )
-
+        try:
+            cluster_estimations = cluster_estimation_to_main_queue.queue.get_nowait()
+        except queue.Empty:
+            cluster_estimations = None
+        if cluster_estimations is not None:
+            for cluster in cluster_estimations:
+                main_logger.debug("Cluser in world: True")
+                main_logger.debug("Cluster location x: " + str(cluster.location_x))
+                main_logger.debug("Cluster location y: " + str(cluster.location_y))
+                main_logger.debug("Cluster spherical variance: " + str(cluster.spherical_variance))
```

Comment:
Rather than parsing the fields of the object here, I think its better to override the `__str__` method for that class. You can see `detection_in_world` as an example and do the same for `object_in_world`. Then in here you can just log the object in one line. To see an example of that look at `geolocation.py line 320`.